### PR TITLE
  Bear.py: Promotes log entry from warning to error 

### PR DESCRIPTION
**Pull request by palash25**
GitCommitBear fails but the exit status is not set. This PR solves this problem by promoting the warning to error

Fixes: #4567

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by palash25** [comment]
Can anyone tell me why are the CI tests failing and how can I improve this PR.

**Comment by jayvdb** [comment]
Five tests are failing.  They can be found in the Details of the build jobs. (and seen on your own computer by running the tests)

```
FAIL tests/bears/BearTest.py::BearTest::test_bad_bear
FAIL tests/bears/BearTest.py::BearTest::test_print_filename_LocalBear
FAIL tests/bears/BearTest.py::BearTest::test_print_no_filename_GlobalBear
FAIL tests/processes/BearRunningTest.py::BearRunningUnitTest::test_strange_bear
FAIL tests/processes/BearRunningTest.py::BearRunningIntegrationTest::test_run
```

**Comment by rhemon** [comment]
since tests are failing I think you will need to edit the test file to suit the changes you have done.

File: tests/processes/BearRunningTest.py [hunk 1]
```diff
@@ -215,7 +215,6 @@ def test_strange_bear(self):
                              LOG_LEVEL.DEBUG,
                              LOG_LEVEL.DEBUG,
                              LOG_LEVEL.WARNING]
-
```
Comment (Makman2) [inline tests/processes/BearRunningTest.py]: unnecessary change

**Comment by Makman2** [comment]
If your commit addresses an issue, please add a reference :+1:
Also regarding your current commit body:
You don't have to mention that you are adjusting tests. This is usual, so you don't have to mention it explicitly again ;) (So imo you can just remove the current commit body altogether, except the issue reference you will probably add)

**Comment by Makman2** [comment]
Although gitmate hasn't detected this in your shortlog: `Promotes` -> `Promote`, we use imperative tense :+1:

File: tests/bears/.coafile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[cli]
+files = BearTest.py
+bears = spaceconsistencybear
```
Comment (Monal5031) [inline tests/bears/.coafile]: Shouldn't this be `SpaceConsistencyBear` ?

Comment (palash25) [inline tests/bears/.coafile]: Done.

File: tests/processes/BearRunningTest.py [hunk 3]
```diff
@@ -214,8 +214,7 @@ def test_strange_bear(self):
                              LOG_LEVEL.ERROR,
                              LOG_LEVEL.DEBUG,
                              LOG_LEVEL.DEBUG,
-                             LOG_LEVEL.WARNING]
-
```
Comment (Makman2) [inline tests/processes/BearRunningTest.py]: removing this newline is unnecessary :)

Comment (palash25) [inline tests/processes/BearRunningTest.py]: Gitmate review was failing because of that blank line so I removed it.

**Comment by Nosferatul** [comment]
please run `coala` in your local repo

**Comment by palash25** [comment]
@Makman2 added the newline

File: tests/processes/BearRunningTest.py [hunk 4]
```diff
@@ -280,6 +279,7 @@ def test_run(self):
                              LOG_LEVEL.DEBUG,
                              LOG_LEVEL.DEBUG,
                              LOG_LEVEL.WARNING]
+
```
Comment (Makman2) [inline tests/processes/BearRunningTest.py]: unnecessary change ;)

File: tests/processes/BearRunningTest.py (hunk 3 above)
Comment (Makman2) [inline tests/processes/BearRunningTest.py]: I think you just had trailing whitespaces on that line. If possible please readd it again :)

File: tests/processes/BearRunningTest.py (hunk 4 above)
Comment (palash25) [inline tests/processes/BearRunningTest.py]: I don't get it you asked me not to remove the newline. Do you want me to remove the newline now because that is what I had initially done?

Comment (Makman2) [inline tests/processes/BearRunningTest.py]: sorry didn't want to confuse you. As you see you just add a newline in the "nowhere", meaning adding or removing such newlines is not necessary for your change. So please don't remove any existing empty lines or add them :)

File: tests/bears/.coafile (hunk 2 above)
Comment (yash-nisar) [inline tests/bears/.coafile]: Any reason for this `.coafile` ? We already have one.

File: tests/bears/.coafile [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[cli]
+files = BearTest.py
+bears = SpaceConsistencyBear
```
Comment (Makman2) [inline tests/bears/.coafile]: hm yeah what is this file actually for? :D

Comment (palash25) [reply tests/bears/.coafile]: I think it got created when I ran coala on BearTest.py, I completely forgot about that. I'll delete it then push again.

**Comment by Makman2** [comment]
ack 41caefc

**Comment by Makman2** [comment]
rebase and lgtm ;)

**Comment by palash25** [comment]
@Makman2 it says here update branch. Should I do that and what's the significance of that?

**Comment by Makman2** [comment]
No it's not the same, GitHub will do a merge commit. Please manually rebase and repush :+1:

**Comment by palash25** [comment]
Reabsed and repushed. Please merge.

**Comment by yash-nisar** [comment]
Maybe triggering a rebuild should solve your problem ?

**Comment by palash25** [comment]
@yash-nisar how do I trigger a rebuild ? if I push again it says everything is up to date.

**Comment by jayvdb** [comment]
I've triggered a rebuild.

**Comment by yash-nisar** [comment]
reack 091d076e71686ea8171f09899374b375d06c2068

**Comment by jayvdb** [comment]
@rultor merge
